### ISSUE TITLE
openmc::write_source_bank changed syntax

### ISSUE THREAD

**Issue by lewisgross1296** [neams-th-coe/cardinal#683]
Currently in `OpenMCProblemBase.C` we define this method
```
void
OpenMCProblemBase::writeSourceBank(const std::string & filename)
{
  hid_t file_id = openmc::file_open(filename, 'w', true);
  openmc::write_attribute(file_id, "filetype", "source");
  openmc::write_source_bank(file_id, false);
  openmc::file_close(file_id);
}

```
However, the syntax of `openmc::write_source_bank` changed via this [PR](https://github.com/openmc-dev/openmc/pull/2451) from `void write_source_bank(hid_t group_id, bool surf_source_bank);` to `void write_source_bank(hid_t group_id, gsl::span<SourceSite> source_bank,  const vector<int64_t>& bank_index)`.

I will look into how to update our method and make a PR to resolve this issue. The call in `OpenMCProblemBase` is the only place in Cardinal we call `openmc::write_source_bank`, so this should be straightforward.

**Comment by lewisgross1296** [comment]
I gave this an attempt, following along the syntax of OpenMC's `state_point.cpp` and `source.cpp`, which [both call](https://github.com/search?q=repo%3Aopenmc-dev%2Fopenmc%20write_source_bank&type=code) `write_source_bank`. Both use the syntax
```
write_source_bank(file_id, simulation::source_bank, simulation::work_index);
```
I don't think I can quite exactly copy this syntax without other changes, though, since I get the error messages: `error: ‘source_bank’ is not a member of ‘openmc::simulation’` and `error: ‘work_index’ is not a member of ‘openmc::simulation’`.

I noticed other places in Cardinal use `openmc::simulation::MEMBER`. It looks like I also need to update Cardinal to recognize these members of the `simulation` class, which exist in OpenMC. When I grep'd for `openmc::simulation` only these [two source files](https://github.com/search?q=repo%3Aneams-th-coe%2Fcardinal+openmc%3A%3Asimulation&type=code) came up, so I'm having a hard time finding where the members of `openmc::simulation` that Cardinal can access are defined.

Is this something I need to update in the OpenMC submodule itself? I've never done that before, so not exactly sure. @aprilnovak
